### ISSUE TITLE
Can the podspec version be increased when transitive dependencies change?

### ISSUE THREAD

**Issue by mnespor** [googleads/googleads-mobile-flutter#753]
Hi there! I have an add-to-app Flutter widget that builds on my development machine, but not on CI. I can trace the problem to this commit: https://github.com/googleads/googleads-mobile-flutter/commit/23c37303e542ca7ea16d892a3b06bdc5fa41f0e4

My best guess is that Cocoapods assumes it can cache podspecs locally until their version numbers change.

## Plugin Version

2.0.1

## Steps to Reproduce

Given an iPhone app with a Flutter add-to-app widget,

1. Declare a dependency on Google Mobile Ads in `Podfile`: `'Google-Mobile-Ads-SDK', '~> 9.6'`
2. Declare a dependency on google_mobile_ads in pubspec.yaml: `google_mobile_ads: ^2.0.1`
3. `pod install`
4. Wait a few weeks (until Google modifies `google_mobile_ads.podspec` again)
5. `pod install`

**Expected results:**
`Podfile.lock` reflects a new version of `google_mobile_ads`, e.g.:
```
- google_mobile_ads (0.0.2):
    - Flutter
    - Google-Mobile-Ads-SDK (= 9.13.0)
```

**Actual results:**
`pod install` fails to update this block of `Podfile.lock` from 9.6.0 to 9.13, possibly because the `google_mobile_ads` version number remained on `0.0.1`.

```
- google_mobile_ads (0.0.1):
    - Flutter
    - Google-Mobile-Ads-SDK (= 9.6.0)
```
This leads to a build failure:
```
[!] CocoaPods could not find compatible versions for pod "Google-Mobile-Ads-SDK":
  In snapshot (Podfile.lock):
    Google-Mobile-Ads-SDK (= 9.6.0, >= 8.0, ~> 9.6)

  In Podfile:
    gadzooks (from `Modules/gadzooks`) was resolved to 6.5.0, which depends on
      Google-Mobile-Ads-SDK (~> 9.6)

    google_mobile_ads (from `Modules/native-apps-flutter/games/games_module/.ios/.symlinks/plugins/google_mobile_ads/ios`) was resolved to 0.0.1, which depends on
      Google-Mobile-Ads-SDK (~> 9.13)
```

**Comment by darshankawar** [comment]
Thanks for the report @mnespor 
This looks similar to https://github.com/googleads/googleads-mobile-flutter/issues/673 which was fixed recently.
Also, the plugin version you are on is not latest, maybe you can try to upgrade the version to [latest](https://pub.dev/packages/google_mobile_ads/changelog#230) and run your use case once again to see if the behavior still persists or not.

**Comment by mnespor** [comment]
Hi @darshankawar , thanks for the review!

#673 is unrelated as far as I can tell. It looks like #673 was an incompatibility between two podspecs. #753 is more so about an incompatibility within one podspec over time, possibly due to a violation of semver in an ecosystem that expects semver.

The key thing that's happening on the CI machine is that I have two ranges: `Google-Mobile-Ads-SDK, ~> 9.6` and `google_mobile_ads: ^2.0.1`. `pod` could satisfy those constraints with `2.3.0` and `9.13`, but something is going wrong, and it's not finding that resolution.

I have two guesses:

- It's possible `pod` assumes a one-to-one relationship between `google_mobile_ads` podspec versions and `Google-Mobile-Ads-SDK` versions, and it cached the 0.0.1->9.6.0 relationship. When 0.0.1 was later republished with ~>9.11 (one to many), maybe `pod` failed to recognize the change.
- It's possible `install_all_flutter_pods` didn't recognize the incompatibility, and didn't update `google_mobile_ads` from 2.0.1 to 2.3.0 in `pubspec.lock`.

**Comment by mnespor** [comment]
I think this might be fixed already, actually? Sorry for the bother.

*What worked*
- I switched from Flutter 3.3 to Flutter@master. On master, `pod install` did update `pubspec.lock` to include google_mobile_ads@2.3.0.

*Things I tried that didn't work*
- `flutter clean && flutter pub get`
- `pod cache clean google_mobile_ads`
- `rm Podfile.lock`

At this point, the second guess – `install_all_flutter_pods` – seems most likely to me. There might have been a behavior in `install_all_flutter_pods` that changed between 3.3 and `master`?
